File: app/src/main/java/com/sky/chowder/model/ApiResponse.java
/*
 * Copyright (C) 2015. Keegan小钢（http://keeganlee.me）
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package com.sky.chowder.model;

/**
 * Created by sky on 15/12/9 下午8:54.
 * Api响应结果的封装类.
 */
public class ApiResponse<T> {
    private String event;    // 返回状态码,"0"代表成功
    private String message;  // 返回信息
    private T obj;           // 单个对象
    private T objList;       // 数组对象
    private int currentPage; // 当前页数
    private int pageSize;    // 每页显示数量
    private int maxCount;    // 总条数
    private int maxPage;     // 总页数


    public ApiResponse(String event, String message) {
        this.event = event;
        this.message = message;
    }

    public boolean isSuccess() {
        return event.equals("0");
    }

    public String getEvent() {
        return event;
    }

    public void setEvent(String event) {
        this.event = event;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public T getObj() {
        return obj;
    }

    public void setObj(T obj) {
        this.obj = obj;
    }

    public T getObjList() {
        return objList;
    }

    public void setObjList(T objList) {
        this.objList = objList;
    }

    public int getCurrentPage() {
        return currentPage;
    }

    public void setCurrentPage(int currentPage) {
        this.currentPage = currentPage;
    }

    public int getPageSize() {
        return pageSize;
    }

    public void setPageSize(int pageSize) {
        this.pageSize = pageSize;
    }

    public int getMaxCount() {
        return maxCount;
    }

    public void setMaxCount(int maxCount) {
        this.maxCount = maxCount;
    }

    public int getMaxPage() {
        return maxPage;
    }

    public void setMaxPage(int maxPage) {
        this.maxPage = maxPage;
    }
}

File: app/src/main/java/com/sky/chowder/model/ApiResponse.kt
/*
 * Copyright (C) 2015. Keegan小钢（http://keeganlee.me）
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package com.sky.chowder.model

/**
 * Created by SKY on 15/12/9 下午8:54.
 * Api响应结果的封装类.
 */
class ApiResponse<T> {
    var event: String? = null    // 返回状态码,"0"代表成功
    var message: String? = null  // 返回信息
    var obj: T? = null           // 单个对象
    var objList: T? = null       // 数组对象
    var currentPage: Int = 0 // 当前页数
    var pageSize: Int = 0    // 每页显示数量
    var maxCount: Int = 0    // 总条数
    var maxPage: Int = 0     // 总页数

    fun ApiResponse(event: String, message: String) {
        this.event = event
        this.message = message
    }

    val isSuccess: Boolean
        get() = event == "0"
}
